gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java

```diff
@@ -66,14 +66,18 @@ protected PolicyPlugin create(Plugin plugin, Class<?> pluginClass) {
 
     private void determineProxyPhases(Class policyClass, PolicyPluginImpl entity) {
         if (!entity.manifest().properties().containsKey("proxy") && !entity.manifest().properties().containsKey("message")) {
-            Set<String> proxyPhases = new HashSet<>();
-            if (methodFoundFor(REQUEST_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("REQUEST");
-            }
-            if (methodFoundFor(RESPONSE_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("RESPONSE");
+            try {
+                Set<String> proxyPhases = new HashSet<>();
+                if (methodFoundFor(REQUEST_ANNOTATIONS, policyClass)) {
+                    proxyPhases.add("REQUEST");
+                }
+                if (methodFoundFor(RESPONSE_ANNOTATIONS, policyClass)) {
+                    proxyPhases.add("RESPONSE");
+                }
+                entity.manifest().properties().put("proxy", proxyPhases.stream().collect(Collectors.joining(",")));
+            } catch (Exception e) {
+                logger.debug("Unable to autodetect the execution phases for the plugin {}.", entity.id(), e);
```

Comment:
I understand the context so you don't log it in warning, maybe add a comment to explain when this happens and why it is only debug logging.